BridgeAppSDK/DataArchiving/SBAActivityArchive.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  SBAActivityArchive.swift
+//  BridgeAppSDK
+//
+//  Created by Erin Mounts on 5/18/16.
+//  Copyright © 2016 Sage Bionetworks. All rights reserved.
+//
+
+private let kSurveyCreatedOnKey               = "surveyCreatedOn";
+private let kSurveyGuidKey                    = "surveyGuid";
+private let kSchemaRevisionKey                = "schemaRevision";
+
+class SBAActivityArchive: SBADataArchive {
+    
+    init(reference: String, result: SBAActivityResult) {
+        super.init(reference: reference)
+        
+        if (result.schemaRevision != nil) {
+            self.setArchiveInfoObject(result.schemaRevision, forKey: kSchemaRevisionKey)
+        }
+        if let surveyReference = result.schedule.activity.survey {
+            // Survey schema is better matched by created date and survey guid
+            self.setArchiveInfoObject(surveyReference.guid, forKey: kSurveyGuidKey)
+            self.setArchiveInfoObject(surveyReference.createdOn.ISO8601String(), forKey: kSurveyCreatedOnKey)
+        }
```

Comment:
Also include the taskIdentifier (if applicable) and the SBBScheduledActivity.guid value so that the session and task (training vs. regular) can be tracked.